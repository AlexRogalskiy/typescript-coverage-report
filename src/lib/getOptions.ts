import { Option } from "commander";
import { ProgramOptions } from "./";

type CommandOption = Option & {
  defaultValue?: string;
};

const ignoredOptions = ["version"];

const getOptionName = (option: CommandOption): string =>
  option.long.replace("--", "");

const getOptions = (program: any) => {
  return (program.options as CommandOption[])
    .filter((option) => !ignoredOptions.includes(getOptionName(option)))
    .reduce((carry, option) => {
      return {
        ...carry,
        [getOptionName(option)]:
          program[getOptionName(option)] || option.defaultValue
      };
    }, {}) as ProgramOptions;
};

export default getOptions;
