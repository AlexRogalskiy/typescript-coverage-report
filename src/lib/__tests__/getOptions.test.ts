import getOptions from "../getOptions";

describe("getOptions function", () => {
  it("returns a list of options with default values", () => {
    const options = getOptions({
      options: [
        {
          long: "--version"
        },
        {
          long: "--outputDir",
          defaultValue: "coverage-ts"
        },
        {
          long: "--strict",
          defaultValue: false
        }
      ],
      outputDir: "coverage-type"
    });

    expect(options).toMatchSnapshot();
  });
});
