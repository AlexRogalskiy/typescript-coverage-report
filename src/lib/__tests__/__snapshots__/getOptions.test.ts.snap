// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getOptions function returns a list of options with default values 1`] = `
Object {
  "outputDir": "coverage-type",
  "strict": false,
}
`;
